src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -204,7 +204,14 @@ void BackgroundNotificationCb (NSNotification obj)
 		void RemoveInflightData (NSUrlSessionTask task, bool cancel = true)
 		{
 			lock (inflightRequestsLock) {
-				inflightRequests.Remove (task);
+				InflightData data;
+				if (inflightRequests.TryGetValue (task, out data)) {
+					data = inflightRequests [task];
```

Comment:
why ? you already requested `data` on the previous line and it was found (positive)

Reply:
Me writing code at 3 am, valid point.